services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -223,9 +223,22 @@ async def _start_dynamic_service(
         node_id=node_uuid,
     )
 
+    user_role: Optional[UserRole] = None
+    try:
+        user_role = await get_user_role(request.app, user_id)
+    except UserNotFoundError:
+        user_role = None
+
+    save_state = await ProjectDBAPI.get_from_app_context(request.app).has_permission(
+        user_id=user_id, project_uuid=f"{project_uuid}", permission="write"
+    )
+    if user_role is None or user_role <= UserRole.GUEST:
```

Comment:
in case user_role is None. is this actually possible??? you do not need to call in the DB right?

Reply:
The user_id always exists, so ti will always have a role here. Good point. I've also redacted this piece of code to make less calls to the db.